### PR TITLE
starting the trajectory Rmd

### PR DESCRIPTION
**Pull request by stephaniehicks**
@LTLA I think I understand how the OSCABase repo works, but I'd welcome your feedback

### CODE REVIEW

File: analysis/trajectory.Rmd [hunk 1]
```diff
@@ -10,8 +10,77 @@ bibliography: ../ref.bib
...
 ```
 
-*WIP*
+In this chapter, we will walk through a minimal analysis to perform lineage reconstruction and psuedotime inference for single-cells along a continuum. 
+We will demonstrate this with the [`slinghot`](https://bioconductor.org/packages/slingshot) Bioconductor package. 
+The goal of slingshot is to use clusters of cells to uncover global structure and convert this structure into smooth lineages represented by one-dimensional variables, called "pseudotime".
+The package offers tools to learn the cluster relationships in an unsupervised or semi-supervised manner and constructing smooth curves representing each lineage, along with visualization methods for each step.
+Slingshot was designed to model developmental trajectories in scRNA-seq data and serve as a component in an analysis pipeline after dimensionality reduction and clustering. 
 
+We will use the 10X PBMC data in the Bioconductor data package `TENxPBMCData` that we have been using in previous chapters.
```
Comment (LTLA) [inline analysis/trajectory.Rmd]: Use `r Biocpkg("TENxPBMCData")` for package references. Same for slingshot above.

File: analysis/trajectory.Rmd [hunk 2]
```diff
@@ -10,8 +10,77 @@ bibliography: ../ref.bib
...
+```{r}
+head(pbmc4k$cluster)
+```
+
+```{r}
+reducedDimNames(pbmc4k)
+```
+
+At this point, we have everything we need to run slingshot. 
+This is a two-step process composed of identifying the global lineage structure with a cluster-based minimum spanning tree (MST) and fitting simultaneous principal curves to describe each lineage.
```
Comment (LTLA) [inline analysis/trajectory.Rmd]: More theoretical details. In particular, I would like to know what the advantages of this method are compared to other approaches. I would also like details on the most important tunable parameters. For example, have a look at the internal debates of pros/cons of various approaches in the clustering chapter.

File: analysis/trajectory.Rmd [hunk 3]
```diff
@@ -10,8 +10,77 @@ bibliography: ../ref.bib
...
+summary(pbmc4k$slingPseudotime_1)
+```
+
+We can also plot the results: 
+
+```{r}
+colors <- colorRampPalette(brewer.pal(11,'Spectral')[-6])(100)
+plotcol <- colors[cut(pbmc4k$slingPseudotime_1, breaks=100)]
+
+plot(reducedDims(pbmc4k)$PCA, col = plotcol, pch=16, asp = 1)
```
Comment (LTLA) [inline analysis/trajectory.Rmd:125]: `reducedDim(pbmc4k, "PCA")` is more efficient. 

But it seems that one could more simply just use `plotPCA(pbmc4k)` and add ggplot lines to it.

**Review by LTLA — COMMENTED** [review]
Looks on track. You need to finish with a bang where you use the pseudo-time to do something interesting, e.g., identify some interesting trajectory based on a gradient of gene expression.

File: analysis/trajectory.Rmd (hunk 3 above)
Comment (stephaniehicks) [reply analysis/trajectory.Rmd:125]: fixed

File: analysis/trajectory.Rmd (hunk 1 above)
Comment (stephaniehicks) [reply analysis/trajectory.Rmd]: fixed

**Comment by stephaniehicks** [comment]
@LTLA so the hold up is that I'm running into some trouble with the `TENxPBMCData` data package. I deleted the cached dataset and downloaded a fresh copy to make sure that wasn't the problem. 

```
library(TENxPBMCData)
pbmc4k <- TENxPBMCData('pbmc4k')
/users/shicks1/.cache/ExperimentHub
  does not exist, create directory? (yes/no): yes
  |======================================================================| 100%

snapshotDate(): 2019-10-18
see ?TENxPBMCData and browseVignettes('TENxPBMCData') for documentation
downloading 1 resources
retrieving 1 resource
  |======================================================================| 100%

loading from cache
  |======================================================================| 100%

  |======================================================================| 100%

HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:
  #000: H5Dio.c line 199 in H5Dread(): can't read data
    major: Dataset
    minor: Read failed
  #001: H5Dio.c line 601 in H5D__read(): can't read data
    major: Dataset
    minor: Read failed
  #002: H5Dchunk.c line 2259 in H5D__chunk_read(): unable to read raw data chunk
    major: Low-level I/O
    minor: Read failed
  #003: H5Dchunk.c line 3624 in H5D__chunk_lock(): data pipeline read failed
    major: Dataset
    minor: Filter operation failed
  #004: H5Z.c line 1301 in H5Z_pipeline(): required filter 'deflate' is not registered
    major: Data filters
    minor: Read failed
  #005: H5PLint.c line 270 in H5PL_load(): search in path table failed
    major: Plugin for dynamically loaded library
    minor: Can't get value
  #006: H5PLpath.c line 604 in H5PL__find_plugin_in_path_table(): search in path /usr/local/hdf5/lib/plugin encountered an error
    major: Plugin for dynamically loaded library
    minor: Can't get value
  #007: H5PLpath.c line 656 in H5PL__find_plugin_in_path(): can't open directory: /usr/local/hdf5/lib/plugin
    major: Plugin for dynamically loaded library
    minor: Can't open directory or file
Error in h5mread(filepath, name, starts = index) :
  H5Dread() returned an error
```

Anytime I load the dataset from the cache, I get the same error. A quick google search didn't really help either. Any thoughts?

**Comment by PeteHaitch** [comment]
This isn't on JHPCE is it @stephaniehicks? I don't think HDF5 is properly working on there currently.

**Comment by stephaniehicks** [comment]
@PeteHaitch -- 😱 😱 😱 It is on JHPCE. Do you have any links to what's going on or who I need to bug to get this fixed? I guess I could try install everything locally in the mean time. 🤔 

@LTLA -- sorry for the confusion. I'll keep you posted on my progressing getting this sorted out.

**Comment by PeteHaitch** [comment]
@stephaniehicks I think you'll need to go the usual Kasper + bit support route. From memory there was some discussion on Slack (but I don't remember if Biostats or JHU Genomics group) and on the bit support mailing list. I know it's also a roadblock for one of Kasper's projects, so that might bump it up the priority list :)

File: analysis/trajectory.Rmd (hunk 2 above)
Comment (stephaniehicks) [reply analysis/trajectory.Rmd]: added

**Pull request closed without merging**